compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp

```diff
@@ -959,7 +959,15 @@ void ConvertToLLVMPass::runOnOperation() {
   //   multiply or add.
   //
   // TODO(bjacob): Use a lowering that uses specific ARM/X86 intrinsics.
-  tosa::populateTosaRescaleToArithConversionPatterns(&patterns);
+  bool use32BitImpl = false;
+  auto variantOp = getExecutableVariantOp(module);
+  if (succeeded(variantOp) && isRISCV(*variantOp)) {
+    // Use the 32-bit lowering for RISC-V if 'zve32x' is specified and there is
+    // no 64-bit integer vector support.
+    use32BitImpl = hasZve32xFeature(*variantOp) && !hasVFeature(*variantOp) &&
+                   !hasZve64xFeature(*variantOp);
```

Comment:
I'm trying to be defensive here because unfortunately we don't have a way to know if the features provided are correct or their equivalences. Consequently, we shouldn't be using the 32-bit lowering if `+zve32x, +v` or `+zve32x, +zve64x` are used even though they shouldn't (I actually used `+zve32x, +zve64x` in my experiments and not even LLVM complained :)).

Reply:
I see. It depends on when the check is done. As shown in the LLVM snippet, `+v` implies `+zve64d` and `+zve32x,+v` would be effectively `+zve64d` after https://github.com/llvm/llvm-project/blob/232c00b/llvm/lib/Support/RISCVISAInfo.cpp#L814, but it may not be the case if the check is on the input `--cpu-features` string.

Reply:
I just check and got `+m,+a,+f,+d,+c,+v,+zve32x,+zve64x` so it seems we are keeping the raw string around without any processing. I'll open an issue for this. Thanks!